### PR TITLE
Gracefully handle type error in `ProcessEnvironment`

### PR DESCRIPTION
**Pull request by breakds**
This PR adds two improvements:

1. Give more information when hitting a type error in`SharedDataBuffer::Write`.
2. Gracefully quit when `_worker` catches an exception.

### CODE REVIEW

File: alf/environments/parallel_environment.cpp [hunk 1]
```diff
@@ -105,13 +105,22 @@ class SharedDataBuffer {
...
   auto arrays = Flatten(nested_array);
   for (size_t j = 0; j < arrays.size(); ++j) {
-    auto buffer = py::buffer(arrays[j]);
-    const py::buffer_info& info = buffer.request();
-    if (info.ndim == 0) {
-      memcpy(GetBuf(slice_id, j), info.ptr, info.itemsize);
-    } else {
-      assert(info.strides[0] * info.shape[0] == (signed)sizes_[j]);
-      memcpy(GetBuf(slice_id, j), info.ptr, info.shape[0] * info.strides[0]);
+    try {
```
Comment (emailweixu) [inline alf/environments/parallel_environment.cpp]: how about put try out of the for loop to save some cost

File: alf/environments/process_environment.py [hunk 2]
```diff
@@ -85,6 +85,10 @@ def _worker(conn,
                 penv.worker()
             except KeyboardInterrupt:
                 penv.quit()
+            except Exception as e:
+                print(e)
+                traceback.print_exc()
```
Comment (emailweixu) [inline alf/environments/process_environment.py]: Is one of these two lines redundant?

Comment (breakds) [reply alf/environments/process_environment.py]: Fixed.

File: alf/environments/parallel_environment.cpp (hunk 1 above)
Comment (breakds) [reply alf/environments/parallel_environment.cpp]: Fixed.

**Review by emailweixu — APPROVED** [review]

**Pull request merged**